dvc/remote/base.py

```diff
@@ -111,6 +113,13 @@ def __init__(self, repo, config):
         self.cache_types = config.get("type") or copy(self.DEFAULT_CACHE_TYPES)
         self.cache_type_confirmed = False
 
+        url = config.get("url")
+        if url:
+            index_name = hashlib.md5(url.encode("utf-8")).hexdigest()
```

Comment:
Let's use sha256. It doesn't really matter, but we've been getting people lurking through our code and saying that the fact that dvc uses md5s doesn't help it to get accepted in the company for security reasons. So we might as well use sha256 to look better here :) Sure, we have a lot of other places where we need to get rid of md5, but it is a good start :slightly_smiling_face: